entry-generation/godot-compiler-plugin/src/main/kotlin/godot/compiler/plugin/CommonComponentRegistrar.kt

```diff
@@ -0,0 +1,66 @@
+package godot.compiler.plugin
+
+import com.intellij.mock.MockProject
+import de.jensklingenberg.mpapt.common.MpAptProject
+import godot.annotation.processor.GodotAnnotationProcessor
+import org.jetbrains.kotlin.codegen.extensions.ClassBuilderInterceptorExtension
+import org.jetbrains.kotlin.compiler.plugin.*
+import org.jetbrains.kotlin.config.CompilerConfiguration
+import org.jetbrains.kotlin.extensions.StorageComponentContainerContributor
+import org.jetbrains.kotlin.js.translate.extensions.JsSyntheticTranslateExtension
+
+class CommonComponentRegistrar : ComponentRegistrar {
```

Comment:
wait, is this required? As I can see it's a duplicate of NativeComponentRegistrar.

Reply:
Yes it's basically the same. The only real difference is what components are registered. We basically only need the Native one but for compiler plugins you need to specify both if you want to use the native one.
this method is required: https://github.com/utopia-rise/godot-kotlin/blob/3ddfee05b972567090ca2b7f3a028aff6607c88d/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/subplugin/GodotSubPlugin.kt#L45

Reply:
Ahhh, last question are those components required to make mpapt work?